lib/holidays/finder/context/parse_options.rb

```diff
@@ -42,12 +42,6 @@ def parse_regions!(regions)
 
           loaded_regions = []
 
-          #FIXME I don't know what this means or why we have it! I'm reluctant to remove it
-          # at this time without understanding more. -PP 2017/3/29
-          #
-          # special case for north_america/US cross-linking
-          load_region!(:north_america) if regions.include?(:us)
-
```

Comment:
This was always loading all of North America when working with `:us` holidays.

Previous behavior:

```ruby
Holidays.on(Date.new(2018, 1, 1), :us)
 => [{:date=>Mon, 01 Jan 2018, :name=>"New Year's Day", :regions=>[:us]}]

Holidays::Factory::Definition.regions_repository.all_loaded
 => [:ca, :ca_qc, :ca_ab, ... , :mx, :mx_pue, :us_fl, :us_la, :us, ...
```

New behavior:

```ruby
Holidays.on(Date.new(2018, 1, 1), :us)
 => [{:date=>Mon, 01 Jan 2018, :name=>"New Year's Day", :regions=>[:us]}]

Holidays::Factory::Definition.regions_repository.all_loaded
 => [:us]
```